lucille-core/src/main/java/com/kmwllc/lucille/stage/Chunking.java

```diff
@@ -0,0 +1,247 @@
+  private final String separator;
+  private final String outputName;
+  private final String text_field;
+  private SentenceDetector sentenceDetector;
+  private static final Logger log = LogManager.getLogger(Chunking.class);
+
+  public Chunking(Config config) throws StageException {
+    super(config, new StageSpec()
+        .withOptionalProperties("chunking_method", "new_chunk_size", "output_name", "separator", "character_limit",
+            "chunks_to_overlap")
+        .withRequiredProperties("text_field"));
+    characterLimit = config.hasPath("character_limit") ? config.getInt("character_limit") : -1;
+    newChunkSize = config.hasPath("new_chunk_size") ? config.getInt("new_chunk_size") : 1;
+    chunksToOverlap = config.hasPath("chunks_to_overlap") ? config.getInt("chunks_to_overlap") : null;
+    method = ChunkingMethod.fromConfig(config);
+    separator = config.hasPath("separator") ? config.getString("separator") : "";
+    text_field = config.getString("text_field");
+    outputName = config.hasPath("output_name") ? config.getString("output_name") : "chunk";
+
+    if (newChunkSize < 1) {
+      throw new StageException("newChunkSize must be greater than 1.");
+    }
+    if (chunksToOverlap != null && chunksToOverlap >= newChunkSize) {
+      throw new StageException("chunksToOverlap must be smaller than newChunkSize.");
+    }
+    if (characterLimit < -1) {
+      throw new StageException("Character limit must be a positive integer");
+    }
+    if (text_field == null) {
+      throw new StageException("text_field configuration where Chunk Stage would take as input is missing");
+    }
+    if (method == ChunkingMethod.CUSTOM && separator.isEmpty()) {
+      throw new StageException("Provide a non empty regEx for separator configuration");
+    }
+    if (method == ChunkingMethod.FIXED && characterLimit <= 0) {
+      throw new StageException("Provide a positive character_limit for fixed sized chunking");
+    }
+  }
+
+  @Override
+  public void start() throws StageException {
+    // load sentence model if we are using that chunking method
+    if (method == ChunkingMethod.SENTENCE) {
+      try (InputStream sentModelIn = getClass().getResourceAsStream("/en-sent.bin")) {
+        if (sentModelIn == null) {
+          throw new StageException("No sentence model found");
+        }
+        SentenceModel sentModel = new SentenceModel(sentModelIn);
+        sentenceDetector = new SentenceDetectorME(sentModel);
+      } catch (IOException e) {
+        throw new StageException("Could not load sentence model", e);
+      }
+    }
+  }
+
+  @Override
+  public Iterator<Document> processDocument(Document doc) throws StageException {
+    // retrieve content to chunk
+    String content = doc.getString(text_field);
+    // for testing: log.info("content retrieved {} ", content);
+
+    // splitting up content based on chunking method
+    String[] chunks;
+    switch (method) {
+      case CUSTOM:
```

Comment:
Add a comment explaining how this switch statement supports the CUSTOM chunking bc this was not initially clear from the code